bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,250 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+def _PyFilter(rdf, func_or_str, *args , extra_args = {} ):
+    """
+    Filters the entries of RDF according to a given condition.
+    Arguments:
+        func_or_str: The condition according to which the RDF is to be filtered. 
+            It can be either a python callable or a c style string.
+        *args:
+            Can be at most two of them.
+            a. The list of columns in  which case the starting arguments of the provided function callable in mapped to the columns in the list.
+                If not provide then it tries maps the name of the parameter to a column name of the RDF. Flags an error if not found.
+            b. The name of the Filter as a string.
+        extra_args: If the function needs any other parameters that are not columns and must be passed explicitly.
+    Returns:
+        self: The filtered instance of the RDF.
+    
+    Examples:
+    1. rdf.Filter("x>y")
+        Filter using an c-style expression
+    2. rdf.Filter(lambda x,y: x>y)
+        Filter using a python callable.
+    3. rdf.Filter(lambda x,y: x>y, ["y", "x"])
+        Filter using a python callable with column list passed explicitly.
```

Comment:
Unclear, make it explicit that passing a list of columns is a mapping to the arguments of the callable.